apt/private/deb_translate_lock.bzl

```
@@ -169,12 +188,16 @@ def _deb_translate_lock_impl(rctx):
             "%s/BUILD.bazel" % (package["name"]),
             _PACKAGE_TEMPLATE.format(
                 target_name = package["name"],
-                data_targets = starlark_codegen_utils.to_dict_attr({
-                    "//:linux_%s" % arch: "//%s/%s" % (package["name"], arch)
+                data_targets = "select(%s)" % starlark_codegen_utils.to_dict_attr({
+                    "//:linux_%s" % arch: "//%s/%s:data" % (package["name"], package["arch"])
+                    for arch in architectures._values
+                }),
+                control_targets = "select(%s)" % starlark_codegen_utils.to_dict_attr({
+                    "//:linux_%s" % arch: "//%s/%s:control" % (package["name"], package["arch"])
                     for arch in architectures._values
                 }),
-                control_targets = starlark_codegen_utils.to_dict_attr({
-                    "//:linux_%s" % arch: "//%s/%s:control" % (package["name"], arch)
+                deps = "select(%s)" % starlark_codegen_utils.to_dict_list_attr({
+                    "//:linux_%s" % arch: ["//%s/%s:deps" % (package["name"], package["arch"])]
```

```suggestion
                    "//:linux_%s" % arch: ["//%s/%s:deps" % (package["name"], package["arch"])]
```

since `:deps` target exclude the package, doesn't this create a broken transitive closure? i am not sure what's the point of `deps` here.

---

The point is to have the package dependencies separate and available, similar to how `:data` is the actual package itself. Then, you have the full set of `:deps` and `:data` available as the `target_name`.

---

Yes, but what i am pointing out is that, `:data` is current package, and `:deps` should be everything the transitive closure of the current package, but it's not the case here, because `:deps` doesn't have the transitive closure of the transitives. So it's missing a turtle when traversing down the closure.  

Or am i misunderstanding this new `:deps` target?

---

`:deps` is exactly the same as what was before in `{deps}`, right? Just "one level up" so that it `select`s the right arch and available as a separate target label.

---

Let me see if this clarifies it more:

```diff
diff --git a/apt/private/package.BUILD.tmpl b/apt/private/package.BUILD.tmpl
index 9c3dd61..8810a96 100644
--- a/apt/private/package.BUILD.tmpl
+++ b/apt/private/package.BUILD.tmpl
@@ -1,17 +1,28 @@
(...)
+
+filegroup(
+    name = "deps",
+    srcs = {deps},
     visibility = ["//visibility:public"],
 )
 
 filegroup(
     name = "{target_name}",
+    srcs = [
+        ":deps",
+        ":data",
+    ],
     visibility = ["//visibility:public"],
-    srcs = {deps} + [":data"]
 )
 ```
 
 That's the change in `package.BUILD.tmpl` and in `_PACKAGE_TEMPLATE`.
 
 Since I haven't changed how `deps` is generated, I don't see how it can be missing anything in the transitive closure :-?
 
 As mentioned, `:deps` is a convenience target that is literally what was before in `{deps}`. Now, also in `_PACKAGE_TEMPLATE` which is "one level up" so `@<REPO>//<PACKAGE>:deps` will point to `@<REPO>//<PACKAGE>/<ARCH>:deps` through a `select`.
 
 Let me know if this makes sense. Thanks!!

---

i understand the change, i still don't understand how's that useful and in what context. deps is literally just all data dependencies of a package, an example might make justify adding it. otherwise it's something we have to support forever.

---

OK, I'll remove it and go back to `rcs = {deps} + [":data"]` then!

---

Ah, I just pushed the change and it breaks... because it's actually "needed", the alias at the top of the repo call the per-arch `:deps` in the "inner" per-arch repos...

I'm going to try to fix it removing `:deps` but I really think it should be added... all that it does is just expose the same stuff that's in the `Dependencies` part of the package. I don't see how "supporting this forever" would be hard :-?

---

OK, I just pointed the `:package_name` target in the upper part of the package repo to the `:arch` target in the "inner" per-arch repos, so no need to have `:deps`.

I still think it would be nice to have, but this way is much simpler and you prefer it so here we go!